### ISSUE TITLE
[Windows Support] changes to Deploy Workload page

### ISSUE THREAD

**Issue by davidnuzik** [rancher/rancher#20119]
Version: Master (v2.3) (5/9/19)

**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
- View "Deploy Workload" page. (Screenshot below)
![image](https://user-images.githubusercontent.com/45179589/57493555-4292bc00-727a-11e9-8a36-8fd775766d02.png)

- Notice the options to "Run all pods for this workload on * nodes only"  is very light gray text making it appear to be disabled.
- Notice that these options are checkmarks. We would like radio buttons instead (We don't believe there are any ways to have agnostic images)
- Notice "Host Port is not supported on Windows nodes." warning shows up even if we have not picked "Run all pods for this workload on Windows nodes only" it should only appear if we pick that.

**Result:**
Improves look of the page. Reduces chances of confusion.

**Other details that may be helpful:**
We are not sure the best way to handle the radio buttons. Maybe we want none to be default (requires no API changes?)  or maybe we want Windows to be default (because if have 1 or more windows nodes on cluster it's probably designed to run windows pods so would probably be preferred). This might need API changes though...

**Comment by loganhz** [comment]
Users might have a manifest image which can run on both linux and windows.
if we use `radio`, how users unselect it?

**Comment by cjellick** [comment]
right, if its radio, you need three options. @vincent99 which is best: mutually exclusive checkboxes or 3 radio buttons?

**Comment by loganhz** [comment]
What's the text should be for the third option? :) @davidnuzik @vincent99

**Comment by cjellick** [comment]
> Notice the options to "Run all pods for this workload on * nodes only" is very light gray text making it appear to be disabled.

I definitely agree that the light grey text makes it look like the option is disabled and should be a different color.

> Notice "Host Port is not supported on Windows nodes." warning shows up even if we have not picked "Run all pods for this workload on Windows nodes only" it should only appear if we pick that.

If you choose the third option (select no check boxes, or the third radio button), you should still theoretically see the warning if you have any windows nodes. Do  we already only conditionally show windows options if there is a window node?

**Comment by davidnuzik** [comment]
Yes we were thinking agnostic docker images are _maybe_ possible so 3 options is okay for these extremely rare situations.

Hmm what about this?
Run all pods for this workload on Linux nodes only
Run all pods for this workload on Windows nodes only
Run all pods for this workload on either Linux or Windows nodes

In my opinion, Windows should be default. Right?

**Comment by davidnuzik** [comment]
@cjellick regarding the Host Port warning: No today it is not conditionally shown. It is shown regardless of what is picked (or nothing picked). If this is not quick and easy (conditionally showing it if pick only deploy to windows or windows+linux) then not a problem for our alpha I think. It's just a nice-to-have thing.

**Comment by davidnuzik** [comment]
It is however only shown if and only if there is 1 or more windows nodes in the cluster.

**Comment by vincent99** [comment]
It should definitely show up only if there are Windows nodes, the choice is pointless if there isn't.

Taking up 2 (or 3) more lines in that area that is currently nicely balanced is pretty unfortunate.  I'd probably switch that row to 3 columns and add a middle column for `Target OS:` `All` `Windows` `Linux`.

![image](https://user-images.githubusercontent.com/753917/57494025-60f9b700-727c-11e9-87b7-96c4db3b18b1.png)

**Comment by vincent99** [comment]
or an `input-group`:

![image](https://user-images.githubusercontent.com/753917/57494298-61df1880-727d-11e9-9325-c8e7e8be1f92.png)

**Comment by cjellick** [comment]
not a blocker, but a nice to have for the alpha

**Comment by loganhz** [comment]
in master

**Comment by cjellick** [comment]
available in master and in `v2.3.0-alpha2`

**Comment by davidnuzik** [comment]
Version: v2.3.0-alpha2

All areas mentioned in the issue description were addressed. The look of the selection has changed and I think this makes more sense:

![image](https://user-images.githubusercontent.com/45179589/57648991-e1216480-757b-11e9-8f9c-6172a437275c.png)

I tested each option carefully with different types of deployments and images. I encountered no issues. If you pick windows the workload will deploy with: 
![image](https://user-images.githubusercontent.com/45179589/57649180-60169d00-757c-11e9-8923-b49beb7b637e.png)

and with linux with: 
![image](https://user-images.githubusercontent.com/45179589/57649155-4bd2a000-757c-11e9-8735-5ba7ec988ce7.png)

and with all it will have none (no scheduling requirements), as expected. I confirmed that "All" is indeed agnostic, Linux and Windows indeed only ever end up on linux/windows nodes respectively.
I made sure the UI/HTTP/API look good and are correct.

I tried editing workloads to ensure my changes would indeed take effect appropriately and that I can scale up/down deployments and such. I ensured during scale up we still only schedule to appropriate nodes (linux if linux, windows if windows, all if all).

Finally, cloning is also working good. When you clone it keeps your OS setting and works without issues. When you add a sidecar there is no option to pick OS (which I think is good) the sidecar will still have the appropriate scheduling requirement (windows if windows, linux if linux). Redeploy, deletion, and etc still work as expected as well.
